crates/bevy_winit/src/lib.rs

```diff
@@ -332,27 +331,36 @@ pub fn winit_runner(mut app: App) {
                     scale_factor,
                     new_inner_size,
                 } => {
-                    let mut dpi_change_events =
-                        app.resources.get_mut::<Events<WindowDpiChanged>>().unwrap();
-
                     let winit_windows = app.resources.get_mut::<WinitWindows>().unwrap();
                     let mut windows = app.resources.get_mut::<Windows>().unwrap();
                     let window_id = winit_windows.get_window_id(winit_window_id).unwrap();
                     let window = windows.get_mut(window_id).unwrap();
-                    if window.use_dpi() {
+                    if let Some(sf) = window.scale_factor_override() {
+                        *new_inner_size = winit::dpi::LogicalSize::new(
+                            window.physical_width(),
+                            window.physical_height(),
+                        )
+                        .to_physical(sf)
```

Comment:
This block doesn't appear to have any effect, if I am reading it right, it updates the `new_inner_size` with  a value that will never be read.

Reply:
I imagine it was intended to recalculate the logical size and update the back-end. If so I believe the calculation would require going from the windows logical size to the pysical size via the reported scale factor, then from the physical size to the user logical size via the scale factor override.

Reply:
This value will be read by winit to change the window size. [Reference](https://docs.rs/winit/0.24.0/winit/event/enum.WindowEvent.html#variant.ScaleFactorChanged)

Reply:
Sorry, the inner_size is already in physical units. On window at least, the window can be resize when crossing into a display with a different scaling factor and resolution, so some action should be taken to update the window and issue notifications.

Reply:
OK my bad, I am still confused by constructing a logical size from a the current physical size, then converting that to physical size. Wouldn't that cause the window size to jump every time a scale factor change is received?

Shouldn't it be something like:

```rust
*new_inner_size = winit::dpi::PhysicalSize::new(
  window.physical_width(),
  window.physical_height(),
);
```

Reply:
The current system does something quite similar. keep in mind that `sf` is the override, and will not change when the true scale factor changes. Creating a logical size and converting it to physical is just a neater way to write this:
```rust
*new_inner_size = winit::dpi::PhysicalSize::new(
  window.physical_width() * sf,
  window.physical_height() * sf,
);
```
(Probably wrong due to types)

Reply:
But, unless I am missing something, you are multiplying the current physical height by the logical to physical height scaling factor and setting that as the new physical size. If the scaling factor override is 1.0 it is a nop. If it is something else, then whenever the user changes their system scaling factor, the window will increase in size by the overridden scaling factor.

Reply:
Yup, that was a bug, and now its gone.